src/libraries/System.Private.CoreLib/src/System/Collections/Generic/HashSet.cs

```diff
@@ -1005,10 +1010,20 @@ private void Resize(int newSize, bool forceNewHashCodes)
         /// Sets the capacity of a <see cref="HashSet{T}"/> object to the actual number of elements it contains,
         /// rounded up to a nearby, implementation-specific value.
         /// </summary>
-        public void TrimExcess()
+        public void TrimExcess() => SetCapacity(Count);
+
+        public void TrimExcess(int capacity)
         {
-            int capacity = Count;
+            ArgumentOutOfRangeException.ThrowIfNegative(capacity);
+            ArgumentOutOfRangeException.ThrowIfLessThan(capacity, _count);
 
+            if (capacity == _count)
```

Comment:
Same question as https://github.com/dotnet/runtime/pull/85877/files#r1238581040. Setting the capacity to equal the current count is the most common use for this API yet it's currently implemented to be a no-op.